File: Group5Capstone/Group5Backend/FoodCartRestEasy/src/main/java/com/capstone/groupfive/FoodCartRestEasy/dto/Purchase.java
package com.capstone.groupfive.FoodCartRestEasy.dto;

import com.capstone.groupfive.FoodCartRestEasy.entity.Address;
import com.capstone.groupfive.FoodCartRestEasy.entity.Customer;
import com.capstone.groupfive.FoodCartRestEasy.entity.Order;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

/*  Data Transfer object class  comprising the object 
  references of entity classes such as 'Order','Customer' and 'Address' */
public class Purchase {
	private Customer customer;
	private Address address;
	private Order order;
	public Customer getCustomer() {
		return customer;
	}
	public void setCustomer(Customer customer) {
		this.customer = customer;
	}
	public Address getAddress() {
		return address;
	}
	public void setAddress(Address address) {
		this.address = address;
	}
	public Order getOrder() {
		return order;
	}
	public void setOrder(Order order) {
		this.order = order;
	}
	public Purchase(Customer customer, Address address, Order order) {
		super();
		this.customer = customer;
		this.address = address;
		this.order = order;
	}
	@Override
	public String toString() {
		return "Purchase [customer=" + customer + ", address=" + address + ", order=" + order + "]";
	}

}

File: Group5Capstone/Group5FrontEnd/food-ordering-system/src/app/model/purchase.ts
import { Address } from "./address";
import { Customer } from "./customer";
import { Order } from "./order";
import { User } from "./user";
//Data Transfer Object class which is mapped to DTO class of spring application
export class Purchase {
    order!:Order;
    customer:Customer;
    address!:Address;
    constructor(customer:Customer,address:Address,order:Order){
        this.customer=customer;
        this.order=order;
        this.address=address;
        
    }


}
